### DISCUSSION TITLE
$record and $state always get first record in table

### DISCUSSION THREAD

**Discussion by Darnes1958** [filamentphp/filament#13985] [Help]
### Package

Table builder

### Package Version

v3

### How can we help you?

I re use old database with filament3 project , in my old database I do not name ID in tables as primary key column, I name other name. 
when I use $record or $state in table action it always get the first record in the table , it is not get the curret record .

**Comment by Darnes1958** [comment]
any help please

**Reply by micraux** [reply]
Hi,

First, do you declare your primary key in your model?
[https://laravel.com/docs/11.x/eloquent#primary-keys](https://laravel.com/docs/11.x/eloquent#primary-keys)

**Reply by Darnes1958** [reply]
yes
  protected $connection = 'other';
  protected $guarded = [];
  protected $table = 'main';
  protected $primaryKey ='no';
  public $incrementing = false;
  public $timestamps = false;

**Reply by micraux** [reply]
Ok.

Does the primary key appears in your table if you display it?

```php
Tables\Columns\TextColumn::make('no')
```

**Reply by Darnes1958** [reply]
yes
Tables\Columns\TextColumn::make('no')
                  ->action(function ($state){
                      $this->no=$state;
                      $this->dispatch('takeNo',no: $this->no);
                      $this->dispatch('showMe',no: $this->no);
                  })
                  ->color('primary')
                  ->size(TextColumnSize::ExtraSmall)
                  ->label('الرقم'),

**Comment by Darnes1958** [comment]
I found the problem , because I use custom search inside the query() , it is not wrong to do that , but this is the reason

**Reply by micraux** [reply]
Ok, nice, happy for you!
